ocaml/xapi/pvs_cache_vdi.ml

```diff
@@ -79,11 +79,22 @@ let get_or_recreate_vdi ~__context ~self =
           vdi
   )
 
+let ensure_vdi_is_closed ~__context vdi =
+  (* The pvsproxy daemon normally unmounts and removes the locally attached VBD
+     for the cache VDI. Here we do it as well, just in case pvsproxy did not
+     manage to, for whatever reason. *)
+  try
+    let uuid = Db.VDI.get_uuid ~__context ~self:vdi in
+    let _ = Helpers.call_script !Xapi_globs.pvsproxy_close_cache_vdi [uuid] in
```

Comment:
If this application is partial we would not notice it. A type annotation would catch it.